markups/asciidoc.py

```diff
@@ -0,0 +1,99 @@
+import markups.common as common
+from markups.abstract import AbstractMarkup, ConvertedMarkup
+
+
+class AsciidocMarkup(AbstractMarkup):
+    """Markup class for Asciidoc language.
+    Inherits :class:`~markups.abstract.AbstractMarkup`.
+
+    :param settings_overrides: optional dictionary of overrides for the
+                               `Docutils settings`_
+    :type settings_overrides: dict
+    """
+    name = 'asciidoc'
+    attributes = {
+        common.LANGUAGE_HOME_PAGE: 'https://asciidoc3.org',
+        common.MODULE_HOME_PAGE: 'https://asciidoc3.org/userguide.html',
+        common.SYNTAX_DOCUMENTATION: 'https://asciidoc3.org/userguide.html'
+    }
+
+    file_extensions = ('.adoc', '.asciidoc')
+    default_extension = '.adoc'
+
+    @staticmethod
+    def available():
+        try:
+            importlib.import_module('asciidoc3')
+            importlib.import_module('lxml')
+        except ImportError:
+            return False
+        return True
+
+    def __init__(self, filename=None):
+        AbstractMarkup.__init__(self, filename)
+
+    def convert(self, text):
+        import asciidoc3
+        from asciidoc3 import asciidoc3api
+        from lxml import etree
+        outfile = StringIO()
+        infile = StringIO(text)
+        converter = asciidoc3api.AsciiDoc3API(
+            asciidoc3.__path__[0] + '/asciidoc3.py')
+        try:
+            converter.execute(infile, outfile)
+        except asciidoc3api.AsciiDoc3Error:
+            # Note: asciidoc3 reports/prints these errors for us.
+            # We catch the exception to avoid multiple reports.
+            pass
+        result = outfile.getvalue()
+        result = result.encode()
+        parser = etree.HTMLParser()
+        root_element = etree.fromstring(result, parser)
+        head_element = root_element.xpath('./head')[0]
+        title_element = root_element.xpath('./head/title')[0]
+        style_elements = root_element.xpath('./head/style')
+        javascript_elements = root_element.xpath('./head/script')
+        body_element = root_element.xpath('./body')[0]
+        head = ''
+        for child in head_element.getchildren():
+            head += etree.tostring(
+                child,
+                encoding='unicode',
+                method='html',
+            )
+        body = ''
+        for child in body_element.getchildren():
+            body += etree.tostring(
+                child,
+                encoding='unicode',
+                method='html',
+            )
+        title = title_element.text
+        stylesheet = ''
+        for style_element in style_elements:
+            stylesheet += style_element.text
+        javascript = ''
+        for javascript_element in javascript_elements:
+            javascript += etree.tostring(
+                javascript_element,
+                encoding='unicode',
+                method='html',
+            )
+        return ConvertedAsciidoc(head, body, title, stylesheet, javascript)
```

Comment:
I think you can do this and remove the class below.

```suggestion
        return ConvertedMarkup(body, title, stylesheet, javascript)
```